src/cargo/core/compiler/job_queue/mod.rs

```diff
@@ -952,7 +952,12 @@ impl<'cfg> DrainState<'cfg> {
                 }
 
                 for warning in output.warnings.iter() {
-                    bcx.config.shell().warn(warning)?;
+                    let warning_with_package = format!(
+                        "{} {}",
+                        format!("[{} {}]", unit.pkg.name(), unit.pkg.version()),
```

Comment:
So this is my bad, I think I marked this as "Accepted" without giving this enough thought.

I had also considered whether this actually is needed since we generally don't show warnings for dependencies.  Should we instead cap these and then it becomes less ambiguous?  I don't know the historical context for this but most build scripts are for adapting code to the current system, so the warnings will be relevant to the end user, more than usual.

So assuming we go through with this, I feel like
- We should be brief to not take away from the warning message
- We should be unobtrusive to not take away from the warning message

For any dependency to warn, we likely can't skip on the version.  Maybe we could skip on both if its a workspace member but I'm unsure about the inconsistency.  So I think it can't really be more brief as-is

One way to make this less obtrusive is to do `{name}@{version}:` instead of `[{name} {version}]`.  The `[]` syntax  visually stops the eye.